src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -99,18 +99,51 @@
 
     @Override
     public Connection connect(String url, Properties info) throws SQLException {
+        Pair<MongoConnection, Integer> p = getUnvalidatedConnectionAndTimeout(url, info);
+        Connection conn = p.left();
+        conn.isValid(p.right());
+        return conn;
+    }
+
+    public MongoConnection createUnvalidatedConnection(String url, Properties info)
+        throws SQLException {
+        return getUnvalidatedConnectionAndTimeout(url, info).left();
+    }
+
+    private Pair<MongoConnection, Integer> getUnvalidatedConnectionAndTimeout(String url, Properties info)
+        throws SQLException {
         if (!acceptsURL(url)) {
             return null;
         }
         if (info == null) {
             info = new Properties();
         }
+
         // reuse the code getPropertyInfo to make sure the URI is properly set wrt the passed
         // Properties info value.
         Pair<ConnectionString, DriverPropertyInfo[]> p = getConnectionString(url, info);
+        // ensure that the ConnectionString and Properties are consistent.
+        reconcileProperties(p.right(), info);
+
+        ConnectionString cs = p.left();
+        Connection ret = createDialectConnection(cs, info);
+        System.out.println(cs);
+        // use a timeout of 5s if no timeout is specified in the URL.
+        int timeout = 5;
```

Comment:
From the error message we were getting when ADL did not start, it seems that the Java driver has a default timeout of 30000ms. 
Why not keeping it as is?

I understand that it might for faster fail when testing, but it would save us from having to document the differences between the Java driver default value and the JDBC driver ones if we were not changing these values.
If we want tests to have a shorter timeout, I'm sure that we can set it in the default connection url(s).

Reply:
30 is extremely long, but that makes sense.